### ISSUE TITLE
Buffer overflow in Java_java_net_Inet4AddressImpl_getLocalHostName of OpenJDK6 and OpenJDK 7

### ISSUE THREAD

**Issue by zjhzxhz** [travis-ci/travis-ci#5227]
Here's my `.travis.yml`, everything seems fine except OpenJDK 7:

```
language: java
jdk:
  - oraclejdk8
  - oraclejdk7
  - openjdk7

services: mysql

before_install:
# Install gcc 4.8
  - sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
  - sudo apt-get update -qq
# Fix path error on JNI headers
  - sudo ln -nsf ${JAVA_HOME}/include/linux/jni_md.h ${JAVA_HOME}/include/jni_md.h
  - sudo ln -nsf ${JAVA_HOME}/include/linux/jawt_md.h ${JAVA_HOME}/include/jawt_md.h

install:
  - sudo apt-get install -qq g++-4.8
  - sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-4.8 90

before_script:
  - mysql -e 'CREATE DATABASE test;'
  - mysql test < voj.sql

script: 
  - mvn test -f web/pom.xml
  - mvn test -f judger/pom.xml

after_script:
  - mysql -e 'DROP DATABASE test;'
```

And Here's the log:

```
*** buffer overflow detected ***: /usr/lib/jvm/java-7-openjdk-amd64/jre/bin/java terminated
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(__fortify_fail+0x37)[0x7f7e3343ee57]
/lib/x86_64-linux-gnu/libc.so.6(+0x108d50)[0x7f7e3343dd50]
/usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libnet.so(Java_java_net_Inet4AddressImpl_getLocalHostName+0x190)[0x7f7e1ad7a960]
[0x7f7e29012d98]
======= Memory map: ========
00400000-00401000 r-xp 00000000 08:01 271208                             /usr/lib/jvm/java-7-openjdk-amd64/jre/bin/java
00600000-00601000 r--p 00000000 08:01 271208                             /usr/lib/jvm/java-7-openjdk-amd64/jre/bin/java
00601000-00602000 rw-p 00001000 08:01 271208                             /usr/lib/jvm/java-7-openjdk-amd64/jre/bin/java
02181000-021a2000 rw-p 00000000 00:00 0                                  [heap]
80a00000-81f00000 rw-p 00000000 00:00 0 
81f00000-8b000000 rw-p 00000000 00:00 0 
8b000000-8fe00000 rw-p 00000000 00:00 0 
8fe00000-d9000000 rw-p 00000000 00:00 0 
d9000000-dd500000 rw-p 00000000 00:00 0 
dd500000-100000000 rw-p 00000000 00:00 0 
7f7df4000000-7f7df4021000 rw-p 00000000 00:00 0 
7f7df4021000-7f7df8000000 ---p 00000000 00:00 0 
7f7dfc000000-7f7dfc84a000 rw-p 00000000 00:00 0 
7f7dfc84a000-7f7e00000000 ---p 00000000 00:00 0 
7f7e00000000-7f7e00021000 rw-p 00000000 00:00 0 
7f7e00021000-7f7e04000000 ---p 00000000 00:00 0 
7f7e04000000-7f7e04021000 rw-p 00000000 00:00 0 
7f7e04021000-7f7e08000000 ---p 00000000 00:00 0 
7f7e08000000-7f7e08a13000 rw-p 00000000 00:00 0 
7f7e08a13000-7f7e0c000000 ---p 00000000 00:00 0 
7f7e0c000000-7f7e0c021000 rw-p 00000000 00:00 0 
7f7e0c021000-7f7e10000000 ---p 00000000 00:00 0 
7f7e10000000-7f7e10021000 rw-p 00000000 00:00 0 
7f7e10021000-7f7e14000000 ---p 00000000 00:00 0 
7f7e14000000-7f7e14021000 rw-p 00000000 00:00 0 
7f7e14021000-7f7e18000000 ---p 00000000 00:00 0 
7f7e1933e000-7f7e19345000 r-xp 00000000 08:01 3249                       /lib/x86_64-linux-gnu/libnss_dns-2.15.so
7f7e19345000-7f7e19544000 ---p 00007000 08:01 3249                       /lib/x86_64-linux-gnu/libnss_dns-2.15.so
7f7e19544000-7f7e19545000 r--p 00006000 08:01 3249                       /lib/x86_64-linux-gnu/libnss_dns-2.15.so
7f7e19545000-7f7e19546000 rw-p 00007000 08:01 3249                       /lib/x86_64-linux-gnu/libnss_dns-2.15.so
7f7e19555000-7f7e1955d000 r-xp 00000000 08:01 525388                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libmanagement.so
7f7e1955d000-7f7e1975c000 ---p 00008000 08:01 525388                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libmanagement.so
7f7e1975c000-7f7e1975d000 r--p 00007000 08:01 525388                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libmanagement.so
7f7e1975d000-7f7e1975e000 rw-p 00008000 08:01 525388                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libmanagement.so
7f7e1975e000-7f7e19a4d000 rw-p 00000000 00:00 0 
7f7e19a4d000-7f7e19a89000 r-xp 00000000 08:01 3360                       /lib/x86_64-linux-gnu/libpcre.so.3.12.1
7f7e19a89000-7f7e19c88000 ---p 0003c000 08:01 3360                       /lib/x86_64-linux-gnu/libpcre.so.3.12.1
7f7e19c88000-7f7e19c89000 r--p 0003b000 08:01 3360                       /lib/x86_64-linux-gnu/libpcre.so.3.12.1
7f7e19c89000-7f7e19c8a000 rw-p 0003c000 08:01 3360                       /lib/x86_64-linux-gnu/libpcre.so.3.12.1
7f7e19c8a000-7f7e19c91000 r-xp 00000000 08:01 45453                      /usr/lib/x86_64-linux-gnu/libffi.so.6.0.0
7f7e19c91000-7f7e19e90000 ---p 00007000 08:01 45453                      /usr/lib/x86_64-linux-gnu/libffi.so.6.0.0
7f7e19e90000-7f7e19e91000 r--p 00006000 08:01 45453                      /usr/lib/x86_64-linux-gnu/libffi.so.6.0.0
7f7e19e91000-7f7e19e92000 rw-p 00007000 08:01 45453                      /usr/lib/x86_64-linux-gnu/libffi.so.6.0.0
7f7e19e92000-7f7e19eaa000 r-xp 00000000 08:01 3405                       /lib/x86_64-linux-gnu/libresolv-2.15.so
7f7e19eaa000-7f7e1a0aa000 ---p 00018000 08:01 3405                       /lib/x86_64-linux-gnu/libresolv-2.15.so
7f7e1a0aa000-7f7e1a0ab000 r--p 00018000 08:01 3405                       /lib/x86_64-linux-gnu/libresolv-2.15.so
7f7e1a0ab000-7f7e1a0ac000 rw-p 00019000 08:01 3405                       /lib/x86_64-linux-gnu/libresolv-2.15.so
7f7e1a0ac000-7f7e1a0ae000 rw-p 00000000 00:00 0 
7f7e1a0ae000-7f7e1a0cb000 r-xp 00000000 08:01 3390                       /lib/x86_64-linux-gnu/libselinux.so.1
7f7e1a0cb000-7f7e1a2ca000 ---p 0001d000 08:01 3390                       /lib/x86_64-linux-gnu/libselinux.so.1
7f7e1a2ca000-7f7e1a2cb000 r--p 0001c000 08:01 3390                       /lib/x86_64-linux-gnu/libselinux.so.1
7f7e1a2cb000-7f7e1a2cc000 rw-p 0001d000 08:01 3390                       /lib/x86_64-linux-gnu/libselinux.so.1
7f7e1a2cc000-7f7e1a2cd000 rw-p 00000000 00:00 0 
7f7e1a2cd000-7f7e1a2d0000 r-xp 00000000 08:01 45440                      /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0.3200.4
7f7e1a2d0000-7f7e1a4cf000 ---p 00003000 08:01 45440                      /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0.3200.4
7f7e1a4cf000-7f7e1a4d0000 r--p 00002000 08:01 45440                      /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0.3200.4
7f7e1a4d0000-7f7e1a4d1000 rw-p 00003000 08:01 45440                      /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0.3200.4
7f7e1a4d1000-7f7e1a5c3000 r-xp 00000000 08:01 3258                       /lib/x86_64-linux-gnu/libglib-2.0.so.0.3200.4
7f7e1a5c3000-7f7e1a7c3000 ---p 000f2000 08:01 3258                       /lib/x86_64-linux-gnu/libglib-2.0.so.0.3200.4
7f7e1a7c3000-7f7e1a7c4000 r--p 000f2000 08:01 3258                       /lib/x86_64-linux-gnu/libglib-2.0.so.0.3200.4
7f7e1a7c4000-7f7e1a7c5000 rw-p 000f3000 08:01 3258                       /lib/x86_64-linux-gnu/libglib-2.0.so.0.3200.4
7f7e1a7c5000-7f7e1a7c6000 rw-p 00000000 00:00 0 
7f7e1a7c6000-7f7e1a813000 r-xp 00000000 08:01 45462                      /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.3200.4
7f7e1a813000-7f7e1aa13000 ---p 0004d000 08:01 45462                      /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.3200.4
7f7e1aa13000-7f7e1aa14000 r--p 0004d000 08:01 45462                      /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.3200.4
7f7e1aa14000-7f7e1aa15000 rw-p 0004e000 08:01 45462                      /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.3200.4
7f7e1aa15000-7f7e1ab5d000 r-xp 00000000 08:01 45452                      /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0.3200.4
7f7e1ab5d000-7f7e1ad5c000 ---p 00148000 08:01 45452                      /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0.3200.4
7f7e1ad5c000-7f7e1ad60000 r--p 00147000 08:01 45452                      /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0.3200.4
7f7e1ad60000-7f7e1ad62000 rw-p 0014b000 08:01 45452                      /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0.3200.4
7f7e1ad62000-7f7e1ad64000 rw-p 00000000 00:00 0 
7f7e1ad68000-7f7e1ad69000 rw-p 00000000 00:00 0 
7f7e1ad69000-7f7e1ad73000 r--s 000ae000 08:01 272804                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/resources.jar
7f7e1ad73000-7f7e1ad8a000 r-xp 00000000 08:01 525413                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libnet.so
7f7e1ad8a000-7f7e1af8a000 ---p 00017000 08:01 525413                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libnet.so
7f7e1af8a000-7f7e1af8b000 r--p 00017000 08:01 525413                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libnet.so
7f7e1af8b000-7f7e1af8c000 rw-p 00018000 08:01 525413                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libnet.so
7f7e1af8c000-7f7e1af9c000 r-xp 00000000 08:01 525380                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libnio.so
7f7e1af9c000-7f7e1b19c000 ---p 00010000 08:01 525380                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libnio.so
7f7e1b19c000-7f7e1b19d000 r--p 00010000 08:01 525380                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libnio.so
7f7e1b19d000-7f7e1b19e000 rw-p 00011000 08:01 525380                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libnio.so
7f7e1b19e000-7f7e1b1a1000 r--s 00007000 08:01 1313011                    /home/travis/.m2/repository/org/apache/maven/surefire/surefire-junit4/2.12.4/surefire-junit4-2.12.4.jar
7f7e1b1a1000-7f7e1b1a7000 r--s 00047000 08:01 1312518                    /home/travis/.m2/repository/org/jsoup/jsoup/1.8.2/jsoup-1.8.2.jar
7f7e1b1a7000-7f7e1b1aa000 r--s 00015000 08:01 1312516                    /home/travis/.m2/repository/javax/servlet/javax.servlet-api/3.1.0/javax.servlet-api-3.1.0.jar
7f7e1b1aa000-7f7e1b1ab000 r--s 0000a000 08:01 1312513                    /home/travis/.m2/repository/asm/asm/3.3.1/asm-3.3.1.jar
7f7e1b1ab000-7f7e1b1ad000 r--s 00010000 08:01 1312512                    /home/travis/.m2/repository/net/minidev/asm/1.0.2/asm-1.0.2.jar
7f7e1b1ad000-7f7e1b1b0000 r--s 0001a000 08:01 1312509                    /home/travis/.m2/repository/net/minidev/json-smart/2.1.0/json-smart-2.1.0.jar
7f7e1b1b0000-7f7e1b1b4000 r--s 0001e000 08:01 1312508                    /home/travis/.m2/repository/com/jayway/jsonpath/json-path/1.2.0/json-path-1.2.0.jar
7f7e1b1b4000-7f7e1b1d2000 r--s 00110000 08:01 1312507                    /home/travis/.m2/repository/org/mockito/mockito-all/1.10.19/mockito-all-1.10.19.jar
7f7e1b1d2000-7f7e1b1df000 r--s 0007a000 08:01 1312503                    /home/travis/.m2/repository/org/springframework/spring-test/4.2.0.RELEASE/spring-test-4.2.0.RELEASE.jar
7f7e1b1df000-7f7e1b1e1000 r--s 00009000 08:01 1312502                    /home/travis/.m2/repository/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar
7f7e1b1e1000-7f7e1b1e9000 r--s 00045000 08:01 1312501                    /home/travis/.m2/repository/junit/junit/4.12/junit-4.12.jar
7f7e1b1e9000-7f7e1b1ec000 r--s 0001e000 08:01 1312497                    /home/travis/.m2/repository/org/apache/logging/log4j/log4j-api/2.1/log4j-api-2.1.jar
7f7e1b1ec000-7f7e1b1fe000 r--s 000b8000 08:01 1312496                    /home/travis/.m2/repository/org/apache/logging/log4j/log4j-core/2.1/log4j-core-2.1.jar
7f7e1b1fe000-7f7e1b201000 r--s 00019000 08:01 1312495                    /home/travis/.m2/repository/commons-pool/commons-pool/1.6/commons-pool-1.6.jar
7f7e1b201000-7f7e1b203000 r--s 00002000 08:01 1312494                    /home/travis/.m2/repository/org/apache/geronimo/specs/geronimo-jta_1.0.1B_spec/1.0.1/geronimo-jta_1.0.1B_spec-1.0.1.jar
7f7e1b203000-7f7e1b205000 r--s 0000b000 08:01 1312489                    /home/travis/.m2/repository/org/apache/activemq/activemq-jms-pool/5.11.1/activemq-jms-pool-5.11.1.jar
7f7e1b205000-7f7e1b212000 r--s 0009b000 08:01 1312487                    /home/travis/.m2/repository/org/apache/activemq/activemq-openwire-legacy/5.11.1/activemq-openwire-legacy-5.11.1.jar
7f7e1b212000-7f7e1b214000 r--s 00003000 08:01 1312483                    /home/travis/.m2/repository/org/apache/geronimo/specs/geronimo-j2ee-management_1.1_spec/1.0.1/geronimo-j2ee-management_1.1_spec-1.0.1.jar
7f7e1b214000-7f7e1b216000 r--s 0000b000 08:01 1312481                    /home/travis/.m2/repository/org/fusesource/hawtbuf/hawtbuf/1.11/hawtbuf-1.11.jar
7f7e1b216000-7f7e1b218000 r--s 00006000 08:01 1312480                    /home/travis/.m2/repository/org/apache/geronimo/specs/geronimo-jms_1.1_spec/1.1.1/geronimo-jms_1.1_spec-1.1.1.jar
7f7e1b218000-7f7e1b22d000 r--s 0010c000 08:01 1312479                    /home/travis/.m2/repository/org/apache/activemq/activemq-client/5.11.1/activemq-client-5.11.1.jar
7f7e1b22d000-7f7e1b23e000 r--s 000f4000 08:01 1312477                    /home/travis/.m2/repository/org/apache/activemq/activemq-broker/5.11.1/activemq-broker-5.11.1.jar
7f7e1b23e000-7f7e1b242000 r--s 0001d000 08:01 1312474                    /home/travis/.m2/repository/org/apache/xbean/xbean-spring/3.18/xbean-spring-3.18.jar
7f7e1b242000-7f7e1b244000 r--s 00027000 08:01 1312471                    /home/travis/.m2/repository/org/apache/activemq/activemq-spring/5.11.1/activemq-spring-5.11.1.jar
7f7e1b244000-7f7e1b248000 r--s 0003c000 08:01 539559                     /home/travis/.m2/repository/commons-lang/commons-lang/2.4/commons-lang-2.4.jar
7f7e1b248000-7f7e1b256000 r--s 0007f000 08:01 1312469                    /home/travis/.m2/repository/commons-collections/commons-collections/3.2.1/commons-collections-3.2.1.jar
7f7e1b256000-7f7e1b25e000 r--s 00066000 08:01 1312467                    /home/travis/.m2/repository/org/apache/velocity/velocity/1.7/velocity-1.7.jar
7f7e1b25e000-7f7e1b262000 r--s 00028000 08:01 1312463                    /home/travis/.m2/repository/org/springframework/spring-context-support/4.2.0.RELEASE/spring-context-support-4.2.0.RELEASE.jar
7f7e1b262000-7f7e1b266000 r--s 0002c000 08:01 1312462                    /home/travis/.m2/repository/javax/mail/javax.mail-api/1.5.2/javax.mail-api-1.5.2.jar
7f7e1b266000-7f7e1b268000 r--s 0000e000 08:01 1312404                    /home/travis/.m2/repository/javax/activation/activation/1.1/activation-1.1.jar
7f7e1b268000-7f7e1b270000 r--s 00084000 08:01 1312459                    /home/travis/.m2/repository/com/sun/mail/javax.mail/1.5.2/javax.mail-1.5.2.jar
7f7e1b270000-7f7e1b279000 r--s 00058000 08:01 1312455                    /home/travis/.m2/repository/taglibs/standard/1.1.2/standard-1.1.2.jar
7f7e1b279000-7f7e1b28a000 r--s 00105000 08:01 1312448                    /home/travis/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.5.0/jackson-databind-2.5.0.jar
7f7e1b28a000-7f7e1b2f5000 r--s 0069d000 08:01 402906                     /usr/lib/jvm/java-7-openjdk-amd64/lib/tools.jar
7f7e1b2f5000-7f7e1b36c000 r--s 00a30000 08:01 1312432                    /home/travis/.m2/repository/org/aspectj/aspectjtools/1.8.5/aspectjtools-1.8.5.jar
7f7e1b36c000-7f7e1b36d000 ---p 00000000 00:00 0 
7f7e1b36d000-7f7e1b46d000 rw-p 00000000 00:00 0                          [stack:2472]
7f7e1b46d000-7f7e1b470000 ---p 00000000 00:00 0 
7f7e1b470000-7f7e1b56e000 rw-p 00000000 00:00 0                          [stack:2471]
7f7e1b56e000-7f7e1b571000 ---p 00000000 00:00 0 
7f7e1b571000-7f7e1b66f000 rw-p 00000000 00:00 0                          [stack:2470]
7f7e1b66f000-7f7e1b672000 ---p 00000000 00:00 0 
7f7e1b672000-7f7e1b770000 rw-p 00000000 00:00 0                          [stack:2469]
7f7e1b770000-7f7e1b773000 ---p 00000000 00:00 0 
7f7e1b773000-7f7e1b871000 rw-p 00000000 00:00 0                          [stack:2468]
7f7e1b871000-7f7e1b9fa000 r--p 00000000 08:01 2508                       /usr/lib/locale/locale-archive
7f7e1b9fa000-7f7e1b9fd000 ---p 00000000 00:00 0 
7f7e1b9fd000-7f7e1bafb000 rw-p 00000000 00:00 0                          [stack:2467]
7f7e1bafb000-7f7e1bafe000 ---p 00000000 00:00 0 
7f7e1bafe000-7f7e1bbfc000 rw-p 00000000 00:00 0                          [stack:2466]
7f7e1bbfc000-7f7e1bbfd000 ---p 00000000 00:00 0 
7f7e1bbfd000-7f7e1c000000 rw-p 00000000 00:00 0                          [stack:2465]
7f7e1c000000-7f7e1c021000 rw-p 00000000 00:00 0 
7f7e1c021000-7f7e20000000 ---p 00000000 00:00 0 
7f7e20000000-7f7e20001000 r--s 00004000 08:01 1312488                    /home/travis/.m2/repository/org/apache/activemq/activemq-pool/5.11.1/activemq-pool-5.11.1.jar
7f7e20001000-7f7e2000b000 r--s 0005c000 08:01 1312454                    /home/travis/.m2/repository/jstl/jstl/1.2/jstl-1.2.jar
7f7e2000b000-7f7e2000f000 r--s 00035000 08:01 1312406                    /home/travis/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.5.0/jackson-core-2.5.0.jar
7f7e2000f000-7f7e20015000 r--s 00048000 08:01 1312445                    /home/travis/.m2/repository/com/alibaba/fastjson/1.1.15/fastjson-1.1.15.jar
7f7e20015000-7f7e20035000 r--s 001b8000 08:01 1312441                    /home/travis/.m2/repository/com/alibaba/druid/1.0.13/druid-1.0.13.jar
7f7e20035000-7f7e20050000 r--s 0012e000 08:01 1312440                    /home/travis/.m2/repository/net/sf/ehcache/ehcache-core/2.6.8/ehcache-core-2.6.8.jar
7f7e20050000-7f7e24000000 rw-p 00000000 00:00 0 
7f7e24000000-7f7e24021000 rw-p 00000000 00:00 0 
7f7e24021000-7f7e28000000 ---p 00000000 00:00 0 
7f7e28000000-7f7e28001000 r--s 00007000 08:01 1312472                    /home/travis/.m2/repository/org/slf4j/slf4j-api/1.7.10/slf4j-api-1.7.10.jar
7f7e28001000-7f7e28003000 r--s 00008000 08:01 1312450                    /home/travis/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.5.0/jackson-annotations-2.5.0.jar
7f7e28003000-7f7e2800c000 r--s 000e2000 08:01 1312444                    /home/travis/.m2/repository/mysql/mysql-connector-java/5.1.34/mysql-connector-java-5.1.34.jar
7f7e2800c000-7f7e28013000 r--s 00067000 08:01 402907                     /usr/lib/jvm/java-7-openjdk-amd64/lib/jconsole.jar
7f7e28013000-7f7e28014000 r--s 00002000 08:01 1312438                    /home/travis/.m2/repository/org/mybatis/caches/mybatis-ehcache/1.0.3/mybatis-ehcache-1.0.3.jar
7f7e28014000-7f7e28022000 r--s 000a1000 08:01 1312434                    /home/travis/.m2/repository/org/mybatis/mybatis/3.2.8/mybatis-3.2.8.jar
7f7e28022000-7f7e28026000 r--s 00019000 08:01 1312431                    /home/travis/.m2/repository/org/aspectj/aspectjrt/1.8.5/aspectjrt-1.8.5.jar
7f7e28026000-7f7e2802f000 r--s 00060000 08:01 1312428                    /home/travis/.m2/repository/org/springframework/spring-websocket/4.2.0.RELEASE/spring-websocket-4.2.0.RELEASE.jar
7f7e2802f000-7f7e28038000 r--s 00052000 08:01 1312426                    /home/travis/.m2/repository/org/springframework/spring-messaging/4.2.0.RELEASE/spring-messaging-4.2.0.RELEASE.jar
7f7e28038000-7f7e2803e000 r--s 0003f000 08:01 1312424                    /home/travis/.m2/repository/org/springframework/spring-jms/4.2.0.RELEASE/spring-jms-4.2.0.RELEASE.jar
7f7e2803e000-7f7e28047000 r--s 0005e000 08:01 1312421                    /home/travis/.m2/repository/org/springframework/spring-jdbc/4.2.0.RELEASE/spring-jdbc-4.2.0.RELEASE.jar
7f7e28047000-7f7e28051000 r--s 00068000 08:01 1312420                    /home/travis/.m2/repository/org/springframework/spring-orm/4.2.0.RELEASE/spring-orm-4.2.0.RELEASE.jar
7f7e28051000-7f7e28059000 r--s 00039000 08:01 1312417                    /home/travis/.m2/repository/org/springframework/spring-tx/4.2.0.RELEASE/spring-tx-4.2.0.RELEASE.jar
7f7e28059000-7f7e2805e000 r--s 0003c000 08:01 1312415                    /home/travis/.m2/repository/org/springframework/spring-expression/4.2.0.RELEASE/spring-expression-4.2.0.RELEASE.jar
7f7e2805e000-7f7e2806f000 r--s 000c3000 08:01 1312414                    /home/travis/.m2/repository/org/springframework/spring-webmvc/4.2.0.RELEASE/spring-webmvc-4.2.0.RELEASE.jar
7f7e2806f000-7f7e28087000 r--s 000f1000 08:01 1312411                    /home/travis/.m2/repository/org/springframework/spring-context/4.2.0.RELEASE/spring-context-4.2.0.RELEASE.jar
7f7e28087000-7f7e28095000 r--s 000a5000 08:01 1312409                    /home/travis/.m2/repository/org/springframework/spring-beans/4.2.0.RELEASE/spring-beans-4.2.0.RELEASE.jar
7f7e28095000-7f7e2809e000 r--s 00050000 08:01 1312405                    /home/travis/.m2/repository/org/springframework/spring-aop/4.2.0.RELEASE/spring-aop-4.2.0.RELEASE.jar
7f7e2809e000-7f7e280ae000 r--s 000ab000 08:01 1312401                    /home/travis/.m2/repository/org/springframework/spring-web/4.2.0.RELEASE/spring-web-4.2.0.RELEASE.jar
7f7e280ae000-7f7e280c3000 r--s 000ec000 08:01 1312399                    /home/travis/.m2/repository/org/springframework/spring-core/4.2.0.RELEASE/spring-core-4.2.0.RELEASE.jar
7f7e280c3000-7f7e28712000 rw-p 00000000 00:00 0 
7f7e28712000-7f7e288c4000 r--s 01d30000 08:01 271244                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/rt.jar
7f7e288c4000-7f7e28db7000 rw-p 00000000 00:00 0 
7f7e28db7000-7f7e29000000 rw-p 00000000 00:00 0 
7f7e29000000-7f7e29270000 rwxp 00000000 00:00 0 
7f7e29270000-7f7e2ca86000 rw-p 00000000 00:00 0 
7f7e2ca86000-7f7e30000000 ---p 00000000 00:00 0 
7f7e30000000-7f7e30002000 r--s 0000a000 08:01 1312437                    /home/travis/.m2/repository/org/mybatis/mybatis-spring/1.2.2/mybatis-spring-1.2.2.jar
7f7e30002000-7f7e30004000 r--s 00000000 08:01 1312407                    /home/travis/.m2/repository/aopalliance/aopalliance/1.0/aopalliance-1.0.jar
7f7e30004000-7f7e30006000 r--s 0000e000 08:01 1312403                    /home/travis/.m2/repository/commons-logging/commons-logging/1.2/commons-logging-1.2.jar
7f7e30006000-7f7e3000a000 r--s 00019000 08:01 1312991                    /home/travis/.m2/repository/org/apache/maven/surefire/surefire-api/2.12.4/surefire-api-2.12.4.jar
7f7e3000a000-7f7e3000c000 r--s 00007000 08:01 1312986                    /home/travis/.m2/repository/org/apache/maven/surefire/surefire-booter/2.12.4/surefire-booter-2.12.4.jar
7f7e3000c000-7f7e30010000 r--s 00034000 08:01 402849                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/ext/sunjce_provider.jar
7f7e30010000-7f7e30012000 r--s 00012000 08:01 402846                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/ext/zipfs.jar
7f7e30012000-7f7e30016000 r--s 0003c000 08:01 402850                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/ext/sunpkcs11.jar
7f7e30016000-7f7e3001a000 r--s 00085000 08:01 402847                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/ext/localedata.jar
7f7e3001a000-7f7e3001b000 ---p 00000000 00:00 0 
7f7e3001b000-7f7e3011b000 rw-p 00000000 00:00 0                          [stack:2464]
7f7e3011b000-7f7e3011c000 ---p 00000000 00:00 0 
7f7e3011c000-7f7e30227000 rw-p 00000000 00:00 0                          [stack:2463]
7f7e30227000-7f7e3026f000 rw-p 00000000 00:00 0 
7f7e3026f000-7f7e3027a000 rw-p 00000000 00:00 0 
7f7e3027a000-7f7e302c2000 rw-p 00000000 00:00 0 
7f7e302c2000-7f7e302e9000 rw-p 00000000 00:00 0 
7f7e302e9000-7f7e30532000 rw-p 00000000 00:00 0 
7f7e30532000-7f7e30555000 rw-p 00000000 00:00 0 
7f7e30555000-7f7e3066a000 rw-p 00000000 00:00 0 
7f7e3066a000-7f7e3066b000 rw-p 00000000 00:00 0 
7f7e3066b000-7f7e30673000 r-xp 00000000 08:01 525384                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libzip.so
7f7e30673000-7f7e30872000 ---p 00008000 08:01 525384                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libzip.so
7f7e30872000-7f7e30873000 r--p 00007000 08:01 525384                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libzip.so
7f7e30873000-7f7e30874000 rw-p 00008000 08:01 525384                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libzip.so
7f7e30874000-7f7e30880000 r-xp 00000000 08:01 3264                       /lib/x86_64-linux-gnu/libnss_files-2.15.so
7f7e30880000-7f7e30a7f000 ---p 0000c000 08:01 3264                       /lib/x86_64-linux-gnu/libnss_files-2.15.so
7f7e30a7f000-7f7e30a80000 r--p 0000b000 08:01 3264                       /lib/x86_64-linux-gnu/libnss_files-2.15.so
7f7e30a80000-7f7e30a81000 rw-p 0000c000 08:01 3264                       /lib/x86_64-linux-gnu/libnss_files-2.15.so
7f7e30a81000-7f7e30a8b000 r-xp 00000000 08:01 3383                       /lib/x86_64-linux-gnu/libnss_nis-2.15.so
7f7e30a8b000-7f7e30c8b000 ---p 0000a000 08:01 3383                       /lib/x86_64-linux-gnu/libnss_nis-2.15.so
7f7e30c8b000-7f7e30c8c000 r--p 0000a000 08:01 3383                       /lib/x86_64-linux-gnu/libnss_nis-2.15.so
7f7e30c8c000-7f7e30c8d000 rw-p 0000b000 08:01 3383                       /lib/x86_64-linux-gnu/libnss_nis-2.15.so
7f7e30c8d000-7f7e30ca4000 r-xp 00000000 08:01 3354                       /lib/x86_64-linux-gnu/libnsl-2.15.so
7f7e30ca4000-7f7e30ea3000 ---p 00017000 08:01 3354                       /lib/x86_64-linux-gnu/libnsl-2.15.so
7f7e30ea3000-7f7e30ea4000 r--p 00016000 08:01 3354                       /lib/x86_64-linux-gnu/libnsl-2.15.so
7f7e30ea4000-7f7e30ea5000 rw-p 00017000 08:01 3354                       /lib/x86_64-linux-gnu/libnsl-2.15.so
7f7e30ea5000-7f7e30ea7000 rw-p 00000000 00:00 0 
7f7e30ea7000-7f7e30eaf000 r-xp 00000000 08:01 3408                       /lib/x86_64-linux-gnu/libnss_compat-2.15.so
7f7e30eaf000-7f7e310ae000 ---p 00008000 08:01 3408                       /lib/x86_64-linux-gnu/libnss_compat-2.15.so
7f7e310ae000-7f7e310af000 r--p 00007000 08:01 3408                       /lib/x86_64-linux-gnu/libnss_compat-2.15.so
7f7e310af000-7f7e310b0000 rw-p 00008000 08:01 3408                       /lib/x86_64-linux-gnu/libnss_compat-2.15.so
7f7e310b0000-7f7e310d9000 r-xp 00000000 08:01 525392                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libjava.so
7f7e310d9000-7f7e312d8000 ---p 00029000 08:01 525392                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libjava.so
7f7e312d8000-7f7e312d9000 r--p 00028000 08:01 525392                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libjava.so
7f7e312d9000-7f7e312db000 rw-p 00029000 08:01 525392                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libjava.so
7f7e312db000-7f7e312e9000 r-xp 00000000 08:01 525408                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libverify.so
7f7e312e9000-7f7e314e8000 ---p 0000e000 08:01 525408                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libverify.so
7f7e314e8000-7f7e314ea000 r--p 0000d000 08:01 525408                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libverify.so
7f7e314ea000-7f7e314eb000 rw-p 0000f000 08:01 525408                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libverify.so
7f7e314eb000-7f7e314f2000 r-xp 00000000 08:01 3345                       /lib/x86_64-linux-gnu/librt-2.15.so
7f7e314f2000-7f7e316f1000 ---p 00007000 08:01 3345                       /lib/x86_64-linux-gnu/librt-2.15.so
7f7e316f1000-7f7e316f2000 r--p 00006000 08:01 3345                       /lib/x86_64-linux-gnu/librt-2.15.so
7f7e316f2000-7f7e316f3000 rw-p 00007000 08:01 3345                       /lib/x86_64-linux-gnu/librt-2.15.so
7f7e316f3000-7f7e31709000 r-xp 00000000 08:01 1740                       /lib/x86_64-linux-gnu/libgcc_s.so.1
7f7e31709000-7f7e31908000 ---p 00016000 08:01 1740                       /lib/x86_64-linux-gnu/libgcc_s.so.1
7f7e31908000-7f7e31909000 r--p 00015000 08:01 1740                       /lib/x86_64-linux-gnu/libgcc_s.so.1
7f7e31909000-7f7e3190a000 rw-p 00016000 08:01 1740                       /lib/x86_64-linux-gnu/libgcc_s.so.1
7f7e3190a000-7f7e31a05000 r-xp 00000000 08:01 3270                       /lib/x86_64-linux-gnu/libm-2.15.so
7f7e31a05000-7f7e31c04000 ---p 000fb000 08:01 3270                       /lib/x86_64-linux-gnu/libm-2.15.so
7f7e31c04000-7f7e31c05000 r--p 000fa000 08:01 3270                       /lib/x86_64-linux-gnu/libm-2.15.so
7f7e31c05000-7f7e31c06000 rw-p 000fb000 08:01 3270                       /lib/x86_64-linux-gnu/libm-2.15.so
7f7e31c06000-7f7e31d08000 r-xp 00000000 08:01 1738                       /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.21
7f7e31d08000-7f7e31f07000 ---p 00102000 08:01 1738                       /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.21
7f7e31f07000-7f7e31f0f000 r--p 00101000 08:01 1738                       /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.21
7f7e31f0f000-7f7e31f11000 rw-p 00109000 08:01 1738                       /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.21
7f7e31f11000-7f7e31f14000 rw-p 00000000 00:00 0 
7f7e31f14000-7f7e32a0c000 r-xp 00000000 08:01 525399                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/server/libjvm.so
7f7e32a0c000-7f7e32c0c000 ---p 00af8000 08:01 525399                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/server/libjvm.so
7f7e32c0c000-7f7e32cad000 r--p 00af8000 08:01 525399                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/server/libjvm.so
7f7e32cad000-7f7e32cd0000 rw-p 00b99000 08:01 525399                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/server/libjvm.so
7f7e32cd0000-7f7e32cfd000 rw-p 00000000 00:00 0 
7f7e32cfd000-7f7e32d13000 r-xp 00000000 08:01 3322                       /lib/x86_64-linux-gnu/libz.so.1.2.3.4
7f7e32d13000-7f7e32f12000 ---p 00016000 08:01 3322                       /lib/x86_64-linux-gnu/libz.so.1.2.3.4
7f7e32f12000-7f7e32f13000 r--p 00015000 08:01 3322                       /lib/x86_64-linux-gnu/libz.so.1.2.3.4
7f7e32f13000-7f7e32f14000 rw-p 00016000 08:01 3322                       /lib/x86_64-linux-gnu/libz.so.1.2.3.4
7f7e32f14000-7f7e32f2c000 r-xp 00000000 08:01 3355                       /lib/x86_64-linux-gnu/libpthread-2.15.so
7f7e32f2c000-7f7e3312b000 ---p 00018000 08:01 3355                       /lib/x86_64-linux-gnu/libpthread-2.15.so
7f7e3312b000-7f7e3312c000 r--p 00017000 08:01 3355                       /lib/x86_64-linux-gnu/libpthread-2.15.so
7f7e3312c000-7f7e3312d000 rw-p 00018000 08:01 3355                       /lib/x86_64-linux-gnu/libpthread-2.15.so
7f7e3312d000-7f7e33131000 rw-p 00000000 00:00 0 
7f7e33131000-7f7e33133000 r-xp 00000000 08:01 3391                       /lib/x86_64-linux-gnu/libdl-2.15.so
7f7e33133000-7f7e33333000 ---p 00002000 08:01 3391                       /lib/x86_64-linux-gnu/libdl-2.15.so
7f7e33333000-7f7e33334000 r--p 00002000 08:01 3391                       /lib/x86_64-linux-gnu/libdl-2.15.so
7f7e33334000-7f7e33335000 rw-p 00003000 08:01 3391                       /lib/x86_64-linux-gnu/libdl-2.15.so
7f7e33335000-7f7e334e9000 r-xp 00000000 08:01 3353                       /lib/x86_64-linux-gnu/libc-2.15.so
7f7e334e9000-7f7e336e8000 ---p 001b4000 08:01 3353                       /lib/x86_64-linux-gnu/libc-2.15.so
7f7e336e8000-7f7e336ec000 r--p 001b3000 08:01 3353                       /lib/x86_64-linux-gnu/libc-2.15.so
7f7e336ec000-7f7e336ee000 rw-p 001b7000 08:01 3353                       /lib/x86_64-linux-gnu/libc-2.15.so
7f7e336ee000-7f7e336f3000 rw-p 00000000 00:00 0 
7f7e336f3000-7f7e33700000 r-xp 00000000 08:01 525410                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/jli/libjli.so
7f7e33700000-7f7e338ff000 ---p 0000d000 08:01 525410                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/jli/libjli.so
7f7e338ff000-7f7e33900000 r--p 0000c000 08:01 525410                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/jli/libjli.so
7f7e33900000-7f7e33901000 rw-p 0000d000 08:01 525410                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/jli/libjli.so
7f7e33901000-7f7e33923000 r-xp 00000000 08:01 3267                       /lib/x86_64-linux-gnu/ld-2.15.so
7f7e33923000-7f7e33924000 r--s 00001000 08:01 1313014                    /home/travis/build/zjhzxhz/voj/web/target/surefire/surefirebooter6867090631263281578.jar
7f7e33924000-7f7e33956000 rw-p 00000000 00:00 0 
7f7e33956000-7f7e33a0c000 rw-p 00000000 00:00 0 
7f7e33a0c000-7f7e33a0f000 ---p 00000000 00:00 0 
7f7e33a0f000-7f7e33b12000 rw-p 00000000 00:00 0                          [stack:2462]
7f7e33b12000-7f7e33b13000 r--s 00002000 08:01 402851                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/ext/dnsns.jar
7f7e33b13000-7f7e33b14000 r--s 00007000 08:01 10109                      /usr/share/java/java-atk-wrapper.jar
7f7e33b14000-7f7e33b17000 r--s 0000f000 08:01 402848                     /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/ext/icedtea-sound.jar
7f7e33b17000-7f7e33b1f000 rw-s 00000000 00:12 24064                      /tmp/hsperfdata_travis/2459
7f7e33b1f000-7f7e33b20000 rw-p 00000000 00:00 0 
7f7e33b20000-7f7e33b21000 r--p 00000000 00:00 0 
7f7e33b21000-7f7e33b23000 rw-p 00000000 00:00 0 
7f7e33b23000-7f7e33b24000 r--p 00022000 08:01 3267                       /lib/x86_64-linux-gnu/ld-2.15.so
7f7e33b24000-7f7e33b26000 rw-p 00023000 08:01 3267                       /lib/x86_64-linux-gnu/ld-2.15.so
7ffdf3e8e000-7ffdf3eaf000 rw-p 00000000 00:00 0                          [stack]
7ffdf3fc8000-7ffdf3fca000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
Aborted (core dumped)
```

See also: https://travis-ci.org/zjhzxhz/voj/jobs/95147354

**Comment by eed3si9n** [comment]
Buffer overflow detected is detected also for OpenJDK 6 using Legacy Precise.
[`[precise-gce]` should be added to the subject](https://blog.travis-ci.com/2015-11-27-moving-to-a-more-elastic-future) if OP is also experiencing it on the new platform.

**Comment by cchantep** [comment]
Same for [me](https://travis-ci.org/ReactiveMongo/ReactiveMongo/jobs/95180084)

**Comment by johnjohndoe** [comment]
Same [here](https://travis-ci.org/johnjohndoe/Brockman/jobs/95335570)

**Comment by nicolas-f** [comment]
Same here https://travis-ci.org/orbisgis/orbisgis/jobs/96074148

**Comment by miguno** [comment]
Sane here: https://travis-ci.org/miguno/kafka-storm-starter/jobs/96865652

**Comment by fabiooshiro** [comment]
+1 https://travis-ci.org/fabiooshiro/plastic-criteria/builds/96879078

**Comment by cchantep** [comment]
That's quite a blocker indeed

**Comment by michel-kraemer** [comment]
Same here
https://travis-ci.org/michel-kraemer/bson4jackson/jobs/96944747

**Comment by BanzaiMan** [comment]
Since `sudo` is available for you in this case, you can use the following workaround:

```yaml
before_install:
  - sudo hostname "$(hostname | cut -c1-63)"
```

**Comment by jeffpeiyt** [comment]
+1 same here: https://travis-ci.org/eBay/parallec/builds/97068299

**Comment by miguno** [comment]
The `sudo` workaround didn't help in my case unfortunately:
https://travis-ci.org/miguno/kafka-storm-starter/jobs/97069159

**Comment by BanzaiMan** [comment]
@miguno Thanks for testing. Perhaps JDK is seeing a different `hostname` value.

**Comment by jgeraerts** [comment]
maybe /etc/hosts needs to be adjusted as well

**Comment by javadev** [comment]
+1

**Comment by BanzaiMan** [comment]
@miguno Could you also try tweaking `/etc/hosts` as @jgeraerts suggested?

**Comment by javadev** [comment]
@BanzaiMan How to modify /etc/hosts from travis script?

**Comment by BanzaiMan** [comment]
```yaml
before_install:
  - cat /etc/hosts # optionally check the content *before*
  - sudo hostname "$(hostname | cut -c1-63)"
  - sed -e "s/^\\(127\\.0\\.0\\.1.*\\)/\\1 $(hostname | cut -c1-63)/" /etc/hosts | sudo tee /etc/hosts
  - cat /etc/hosts # optionally check the content *after*
```

**Comment by miguno** [comment]
@BanzaiMan The combination of updating `hostname` as well as `/etc/hosts` apparently fixed the problem.

Example of a successful Travis job with the [workaround](https://github.com/travis-ci/travis-ci/issues/5227#issuecomment-165131913) applied:
https://travis-ci.org/miguno/kafka-storm-starter/jobs/97231537

Workaround as mentioned above:

```
before_install:
  - cat /etc/hosts # optionally check the content *before*
  - sudo hostname "$(hostname | cut -c1-63)"
  - sed -e "s/^\\(127\\.0\\.0\\.1.*\\)/\\1 $(hostname | cut -c1-63)/" /etc/hosts | sudo tee /etc/hosts
  - cat /etc/hosts # optionally check the content *after*
```

For reference, here's `/etc/hosts` before the workaround:

```
$ cat /etc/hosts
127.0.0.1       localhost
127.0.1.1       precise64

# host on which base box was built. Erlang services that use Mnesia
# (like RabbitMQ) may create DB directories with this hostname in the name
# during provisioning and if we change hostname later, they will have all
# kinds of obscure issues. MK.
127.0.0.1       nettuno

# backwards compatibility. MK.
127.0.0.1       travis
127.0.0.1       vagrant
```

And here's `/etc/hosts` after the workaround:

```
$ cat /etc/hosts
127.0.0.1       localhost testing-gce-eab4c5ea-5334-4168-aad8-b620d75575a0
127.0.1.1       precise64

# host on which base box was built. Erlang services that use Mnesia
# (like RabbitMQ) may create DB directories with this hostname in the name
# during provisioning and if we change hostname later, they will have all
# kinds of obscure issues. MK.
127.0.0.1       nettuno testing-gce-eab4c5ea-5334-4168-aad8-b620d75575a0

# backwards compatibility. MK.
127.0.0.1       travis testing-gce-eab4c5ea-5334-4168-aad8-b620d75575a0
127.0.0.1       vagrant testing-gce-eab4c5ea-5334-4168-aad8-b620d75575a0
```

**Comment by miguno** [comment]
Thanks @BanzaiMan !

**Comment by jgeraerts** [comment]
can this workaround be fixed in for example jdk_switcher when switching to openjdk? that would bulk fix a lot of builds.

**Comment by BanzaiMan** [comment]
@jgeraerts Thanks for the suggestion. Unfortunately, `jdk_switcher` is baked into the build image, so changing the content on the fly is a little problematic (though it is not impossible). Also, it is not available on all build images while OpenJDK 6 and/or 7 is.

I think it is best to leave the judgment to the users whether or not this workaround is appropriate.

**Comment by miguno** [comment]
@BanzaiMan If the workaround can be integrated out of the box, what about updating the Travis docs for Java and/or its FAQ?

**Comment by joshk** [comment]
@BanzaiMan i think we should bake this into travis-build somehow. Eg. when a certain jdk is selected.

**Comment by jgeraerts** [comment]
@joshk that's why jdk_switcher seemed like a logical place.

**Comment by joshk** [comment]
I understand @jgeraerts, as @BanzaiMan said that isn't ideal right now.

**Comment by BanzaiMan** [comment]
We are looking at a more comprehensive and less invasive solution to this issue. Thank you for your patience.

**Comment by jgeraerts** [comment]
Just as reference, these seem to be the related issues

* http://mail.openjdk.java.net/pipermail/net-dev/2012-July/004603.html
* http://bugs.java.com/bugdatabase/view_bug.do?bug_id=7089443
* http://bugs.java.com/bugdatabase/view_bug.do?bug_id=7112670

**Comment by my-flow** [comment]
+1 for a more comprehensive and less invasive solution.

**Comment by BanzaiMan** [comment]
I'm working on a fix.

Before: https://travis-ci.org/BanzaiMan/Brockman/builds/100188190
After: https://staging.travis-ci.org/BanzaiMan/Brockman/builds/465906

This changes `hostname` to something shorter, and it *could* have unintended consequences.

**Comment by johnjohndoe** [comment]
@BanzaiMan Thanks for the update. Looking forward to a non-invasive solution.

**Comment by rholder** [comment]
This looks like what I'm also seeing here: https://travis-ci.org/rholder/esthree/jobs/103494475

**Comment by schildbach** [comment]
I wonder if this issue is being worked on? The issue persists for almost 2 months now, and it fails thousands of builds.

**Comment by aoberoi** [comment]
@BanzaiMan thanks for the status update a little while ago, any news?

**Comment by BanzaiMan** [comment]
Hi, everyone,

Sorry for the long delay. We have implemented a `hostname` addon, documented in https://docs.travis-ci.com/user/hostname.

Let us know if you have questions. Thank you!

**Comment by johnjohndoe** [comment]
> This addon, available in build environment with `sudo` enabled, allows you to set the `hostname`.

What do I do if `sudo` is **not** enabled for my build environment?

**Comment by BanzaiMan** [comment]
@johnjohndoe Are you seeing the buffer overflow in the container environment? From what I understand, the containers have short enough a `hostname` that this buffer overflow is not a concern, and the addon is not useful.

**Comment by johnjohndoe** [comment]
I verified builds on [two Java](https://travis-ci.org/johnjohndoe/Brockman/builds/108402191) and one [Android project](https://travis-ci.org/johnjohndoe/CampFahrplan/builds/108403665). I confirm the **OpenJDK builds passed**. Thank you!
I can however not tell if these run in a `sudo` enabled environment.

**Comment by BanzaiMan** [comment]
@johnjohndoe https://travis-ci.org/johnjohndoe/Brockman/jobs/108402194 is one with `sudo` enabled.

**Comment by johnjohndoe** [comment]
Can I tell this from *.travis.yml* or the project configuration on travis-ci.org?

**Comment by BanzaiMan** [comment]
@johnjohndoe You use [`sudo` explicitly](https://github.com/johnjohndoe/Brockman/blob/72d67353ed4700d6f78d2eb46f33d3b7387a0081/.travis.yml#L14) in a customizable section, so we send it to a build env that can use it.

**Comment by johnjohndoe** [comment]
So I could optionally add `sudo: true` at the beginning of the script - but it is not necessary since you inspect the configuration file anyways. Is this correct?

**Comment by BanzaiMan** [comment]
@johnjohndoe That is correct.

**Comment by lukepalmer** [comment]
Unfortunately the hostname: localhost doesn't appear to work for me on OpenJDK7.  Same issue. Opened travis-ci/travis-ci#5669

**Comment by cchantep** [comment]
I think this should be reopened. Even with the workaround, the problem still happen with OpenJDK7.

**Comment by BanzaiMan** [comment]
@cchantep That issue is #5669.

**Comment by viktorklang** [comment]
We're still seeing this issue even after experimenting with many different approaches:

https://travis-ci.org/reactive-streams/reactive-streams-jvm/jobs/120934549

Please advise

**Comment by BanzaiMan** [comment]
@viktorklang Yours is also #5669.

**Comment by PascalSchumacher** [comment]
Even-though this and  https://travis-ci.org/mockito/mockito/jobs/138744704 #5669 are fixed we still got this exception today with openjdk-6: https://travis-ci.org/mockito/mockito/jobs/138744704

Adding the workaround from https://github.com/travis-ci/travis-ci/issues/5227#issuecomment-165131913 fixed the build.

**Comment by bric3** [comment]
This still fail : 

OpenJDK7 
https://travis-ci.org/mockito/mockito/jobs/151920806#L223

```
/lib/x86_64-linux-gnu/libc.so.6(__fortify_fail+0x37)[0x7f534ddc0e37]
/lib/x86_64-linux-gnu/libc.so.6(+0x108d30)[0x7f534ddbfd30]
/usr/lib/jvm/java-7-openjdk-amd64/jre/lib/amd64/libnet.so(Java_java_net_Inet4AddressImpl_getLocalHostName+0x190)[0x7f533b4559a0]
[0x7f5343f8de98]
======= Memory map: ========
00400000-00401000 r-xp 00000000 08:01 669575                             /usr/lib/jvm/java-7-openjdk-amd64/jre/bin/java
00600000-00601000 r--p 00000000 08:01 669575                             /usr/lib/jvm/java-7-openjdk-amd64/jre/bin/java
00601000-00602000 rw-p 00001000 08:01 669575                             /usr/lib/jvm/java-7-openjdk-amd64/jre/bin/java
021ee000-02c25000 rw-p 00000000 00:00 0                                  [heap]
b0000000-b2480000 rw-p 00000000 00:00 0 
b2480000-c0000000 rw-p 00000000 00:00 0 
c0000000-c4e00000 rw-p 00000000 00:00 0 
c4e00000-eaa80000 rw-p 00000000 00:00 0 
eaa80000-fcd80000 rw-p 00000000 00:00 0 
```

OpenJDK6
https://travis-ci.org/mockito/mockito/jobs/151920805#L221

```
/lib/x86_64-linux-gnu/libc.so.6(__fortify_fail+0x37)[0x7f4ee0c80e37]
/lib/x86_64-linux-gnu/libc.so.6(+0x108d30)[0x7f4ee0c7fd30]
/usr/lib/jvm/java-6-openjdk-amd64/jre/lib/amd64/libnet.so(Java_java_net_Inet4AddressImpl_getLocalHostName+0x1a0)[0x7f4ed220bd80]
[0x7f4ed50136b0]
======= Memory map: ========
00400000-00409000 r-xp 00000000 08:01 528113                             /usr/lib/jvm/java-6-openjdk-amd64/jre/bin/java
00608000-00609000 r--p 00008000 08:01 528113                             /usr/lib/jvm/java-6-openjdk-amd64/jre/bin/java
00609000-0060a000 rw-p 00009000 08:01 528113                             /usr/lib/jvm/java-6-openjdk-amd64/jre/bin/java
00712000-01513000 rw-p 00000000 00:00 0                                  [heap]
b0000000-b23f0000 rw-p 00000000 00:00 0 
b23f0000-c0000000 rw-p 00000000 00:00 0 
c0000000-c4e00000 rw-p 00000000 00:00 0 
c4e00000-eaab0000 rw-p 00000000 00:00 0 
eaab0000-fb8b0000 rw-p 00000000 00:00 0 
fb8b0000-100000000 rw-p 00000000 00:00 0 
```

**Comment by BanzaiMan** [comment]
@bric3 Your `/etc/hosts` is empty. https://travis-ci.org/mockito/mockito/jobs/151920806#L199 If you used [my workaround](https://github.com/travis-ci/travis-ci/issues/5227#issuecomment-165131913), you need to update it.

**Comment by bric3** [comment]
@BanzaiMan Thanks a lot. It seems to work.

**Comment by javadev** [comment]
Hi, looks like issue back.

This fix doesn't help anymore:
before_install:
  - sudo hostname "$(hostname | cut -c1-63)"
  - sed -e "s/^\\(127\\.0\\.0\\.1.*\\)/\\1 $(hostname | cut -c1-63)/" /etc/hosts | sudo tee /etc/hosts

Kind regards, Valentyn

**Comment by BanzaiMan** [comment]
@javadev Please read the update to my workaround: https://github.com/travis-ci/travis-ci/issues/5227#issuecomment-165131913.

**Comment by jakirkham** [comment]
Yep, updated workaround does the job. Thanks for the update and nicer syntax. 😄

**Comment by jchambers** [comment]
Folks, this issue was fixed for a while, but seems to have regressed. Workarounds previously mentioned in this thread still work.

**Comment by BanzaiMan** [comment]
@jchambers The hosts addon doesn't work for you? Do you have a build log URL?

**Comment by jchambers** [comment]
@BanzaiMan I really should have included more details in that prior comment. Please pardon the poor form!

- We first added the hostname workaround on April 12, 2016: https://travis-ci.org/relayrides/pushy/builds/122483472
- We reverted the workaround on August 18, 2016: https://travis-ci.org/relayrides/pushy/builds/153582428
- Three days ago (May 5, 2017), we saw builds start failing again under OpenJDK 7 with the same stack trace: https://travis-ci.org/relayrides/pushy/jobs/229370262
- We restored the workaround two days ago (May 6, 2017), which resolved the problem: https://travis-ci.org/relayrides/pushy/builds/229584634

Hope that helps!

**Comment by BanzaiMan** [comment]
Interesting. Could you open a new ticket with this information? We'll take a look.

**Comment by jchambers** [comment]
> Could you open a new ticket with this information?

As requested, I've opened #7732. Thanks for looking into it!
